qiskit_ibm_provider/hub_group_project.py

```diff
@@ -79,19 +79,22 @@ def _discover_remote_backends(self) -> Dict[str, "ibm_backend.IBMBackend"]:
         """
         ret = OrderedDict()
         backends = self._provider._runtime_client.list_backends(self.name)
-        for backend in backends:
-            raw_config = self._provider._runtime_client.backend_configuration(backend)
-            config = configuration_from_server_data(
-                raw_config=raw_config, instance=self.name
-            )
-            if not config:
-                continue
-            ret[config.backend_name] = ibm_backend.IBMBackend(
-                instance=self.name,
-                configuration=config,
-                api_client=self._api_client,
-                provider=self._provider,
-            )
+        if backends:
+            for backend in backends:
+                raw_config = self._provider._runtime_client.backend_configuration(
+                    backend
+                )
+                config = configuration_from_server_data(
+                    raw_config=raw_config, instance=self.name
+                )
+                if not config:
+                    continue
+                ret[config.backend_name] = ibm_backend.IBMBackend(
+                    instance=self.name,
+                    configuration=config,
+                    api_client=self._api_client,
+                    provider=self._provider,
+                )
```

Comment:
The runtime api retrieves backends filtered by hgp while the old iqx endpoint would retrieve all backends the provider has access to [here](https://github.com/Qiskit/qiskit-ibm-provider/blob/6b91c930855bb3c6168c1e8bf450edb37a8e658a/qiskit_ibm_provider/hub_group_project.py#L81) with the configs included. So this block of code shouldn't fail if no backends are in the h/g/p